server/src/main/java/org/opensearch/action/admin/indices/segments/TransportPitSegmentsAction.java

```diff
@@ -0,0 +1,261 @@
+        return new PitSegmentsRequest(in);
+    }
+
+    @Override
+    public List<ShardRouting> getShardRoutingsFromInputStream(StreamInput in) throws IOException {
+        return in.readList(PitAwareShardRouting::new);
+    }
+
+    /**
+     * This retrieves segment details of PIT context
+     * @param request      the node-level request
+     * @param shardRouting the shard on which to execute the operation
+     */
+    @Override
+    protected ShardSegments shardOperation(PitSegmentsRequest request, ShardRouting shardRouting) {
+        if (shardRouting instanceof PitAwareShardRouting) {
+            PitAwareShardRouting pitAwareShardRouting = (PitAwareShardRouting) shardRouting;
+            SearchContextIdForNode searchContextIdForNode = decode(namedWriteableRegistry, pitAwareShardRouting.getPitId()).shards()
+                .get(shardRouting.shardId());
+            PitReaderContext pitReaderContext = searchService.getPitReaderContext(searchContextIdForNode.getSearchContextId());
+            if (pitReaderContext == null) {
+                return new ShardSegments(shardRouting, new ArrayList<>());
+            }
+            return new ShardSegments(pitReaderContext.getShardRouting(), pitReaderContext.getSegments());
+        }
+        return new ShardSegments(shardRouting, new ArrayList<>());
+    }
+
+    /**
+     * This holds PIT id which is used to perform broadcast operation in PIT shards to retrieve segments information
+     */
+    public class PitAwareShardRouting extends ShardRouting {
+
+        private final String pitId;
+
+        public PitAwareShardRouting(StreamInput in) throws IOException {
+            super(in);
+            this.pitId = in.readString();
+        }
+
+        public PitAwareShardRouting(
+            String pitId,
+            ShardId shardId,
+            String currentNodeId,
+            String relocatingNodeId,
+            boolean primary,
+            ShardRoutingState state,
+            RecoverySource recoverySource,
+            UnassignedInfo unassignedInfo,
+            AllocationId allocationId,
+            long expectedShardSize
+        ) {
+            super(
+                shardId,
+                currentNodeId,
+                relocatingNodeId,
+                primary,
+                state,
+                recoverySource,
+                unassignedInfo,
+                allocationId,
+                expectedShardSize
+            );
+            this.pitId = pitId;
+        }
+
+        public String getPitId() {
+            return pitId;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(pitId);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            super.toXContent(builder, params);
+            builder.field("pitId", pitId);
```

Comment:
We have inconsistent field namings for PIT ids:

- `pit_id` in `https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/search/DeletePitRequest.java#L79`
- `id` in `https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/search/CreatePitResponse.java#L31`
- `pitId` here

We should stick to same conventions everywhere.

Reply:
Changed to pit_id